api/grpcserver/grpcserver_test.go

```diff
@@ -2089,6 +2096,15 @@ func TestAccountDataStream_comprehensive(t *testing.T) {
 	wg := sync.WaitGroup{}
```

Comment:
Using a series of waitgroups like this is an antipattern. You should use a (unbuffered) channel instead. The listener reads from the channel, the sender sends into the channel to release it.